### PR TITLE
Fzchriha/labgraph UI

### LINKED ISSUE #45: LabGraph Monitor Improvement - Add input/output message information in LabGraph Monitor UI
## 🚀 Feature
Add input/output message information (e.g. data type and message name) in LabGraph Monitor GUI (frontend)

## Additional context

1. Existing application can be found [here] (https://github.com/facebookresearch/labgraph/pull/38)
2. The code should be added at folder is https://github.com/facebookresearch/labgraph/pull/38
3. Create setup.py and README.md, where example can be found [here](https://github.com/facebookresearch/labgraph/tree/main/extensions/labgraph_viz)
4. Add github action support, reference: [here](https://github.com/facebookresearch/labgraph/actions/workflows/main.yml)
5. Add proper license header, [example](https://github.com/facebook/react/blob/main/scripts/babel/transform-object-assign-require.js).

### PR DESCRIPTION
**Pull request by fzchriha**
## Description

Create UI for real-time messaging and set up logic behind real-time data rendering.

Fixes #45 

## Demo
demo_realtimedata2

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

## Checklist:

- [X] New and existing unit tests pass locally with these changes?
- [X] Has code been commented, particularly in hard-to-understand areas?
- [X] Have you made corresponding changes to the documentation?

### CODE REVIEW

File: extensions/prototypes/labgraph_monitor/public/index.html [hunk 1]
```diff
@@ -1,25 +0,0 @@
...
-      name="description"
-      content="Web site created using create-react-app"
-    />
-    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
-    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
-    <title>Labgraph Monitor</title>
-  </head>
-  <body>
-    <noscript>You need to enable JavaScript to run this app.</noscript>
-    <div id="root"></div>
```
Comment (bennaaym) [inline extensions/prototypes/labgraph_monitor/public/index.html]: Hello @fzchriha, index.html represents the entry point of the SPA, deleting this file will cause **yarn start** / **npm start** to fail when trying to run the application.

Comment (fzchriha) [reply extensions/prototypes/labgraph_monitor/public/index.html]: Hi @bennaaym thank you for the heads up will correct that now

**Comment by fzchriha** [comment]
Here is a preview of what this UI looks like
When you click on a message
Screen Shot 2022-03-18 at 8 22 51 PM

When you click on SHOW REALTIME DATA
Screen Shot 2022-03-18 at 8 22 59 PM

**Comment by fzchriha** [comment]
@jfResearchEng Hi Jimmy I have a question regarding the UX of this design, since my current UI in a way hides the real-time data until it is needed I was wondering what is the predicted **frequency of accessing the real-time data window**?

**Comment by fzchriha** [comment]
I just configured the logic behind real-time messaging, using mock data, please watch the demo below.
demoRealTimeMessage
I will be waiting on @dtemir Damir to finish up [Real-time data on the backend ](https://github.com/facebookresearch/labgraph/pull/64), and then I will update my reducer with the correct data.

**Comment by fzchriha** [comment]
@dtemir the latest commit synchs realtime data in the monitor, thank you for the great docs!

**Comment by jfResearchEng** [comment]
> @jfResearchEng Hi Jimmy I have a question regarding the UX of this design, since my current UI in a way hides the real-time data until it is needed I was wondering what is the predicted **frequency of accessing the real-time data window**?

I would recommend to start with 60Hz and 120Hz.

**Comment by jfResearchEng** [comment]
Closing this for now as PR#71 contains the latest changes.

**Pull request closed without merging**